src/components/MagicCodeInput.js

```diff
@@ -159,17 +159,19 @@ function MagicCodeInput(props) {
         let focusTimeout = null;
         if (props.shouldDelayFocus) {
             focusTimeout = setTimeout(() => inputRefs.current[0].focus(), CONST.ANIMATED_TRANSITION);
+        } else {
+            inputRefs.current[0].focus();
         }
 
-        inputRefs.current[0].focus();
-
         return () => {
             if (!focusTimeout) {
                 return;
             }
             clearTimeout(focusTimeout);
         };
-    }, [props.autoFocus, props.shouldDelayFocus]);
+        // We only want this to run on mount
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
This is aligned with BaseTextInput.
No issue with original dependencies but no need, as auto-focus should happen only on component mount.

Reply:
Done.

Note: mSafari login magic code auto-focus doesn't work on main

Reply:
Not working for me. Tested on iOS 16.4. Might depend on iOS version.


https://github.com/Expensify/App/assets/108292595/2837b5a1-09fc-4ae2-b9ac-aaffafeb1237

It would be great if you can test this branch on your simulator.
As `shouldDelayFocus` is not passed to MagicCodeInput on sign in page, it should not be affected by this PR.